packages/ymax-planner/src/watch-cctp.ts

```diff
@@ -0,0 +1,112 @@
+import { id, zeroPadValue, getAddress, type Provider } from 'ethers';
+
+const TRANSFER = id('Transfer(address,address,uint256)');
+
+type WatchTransferOptions = {
+  provider: Provider;
+  watchAddress: string;
+  expectedAmount: bigint;
+  timeoutMinutes?: number;
+  logPrefix?: string;
+};
+
+const parseTransferLog = log => {
+  if (!log.topics || log.topics.length < 3 || !log.data) {
+    throw new Error('Malformed ERC-20 Transfer log');
+  }
+  return {
+    from: extractAddress(log.topics[1]),
+    to: extractAddress(log.topics[2]),
+    amount: parseAmount(log.data),
+  };
+};
+
+const extractAddress = topic => {
+  // Topics are 32 bytes; Ethereum addresses are last 20 bytes.
+  return getAddress('0x' + topic.slice(-40));
+};
+
+const parseAmount = data => {
+  // ERC-20 value is stored as a 32-byte hex number.
+  return BigInt(data);
+};
+
+export const watchCCTPTransfer = ({
+  provider,
+  watchAddress,
+  expectedAmount,
+  timeoutMinutes = 5,
+  logPrefix = '',
+}: WatchTransferOptions): Promise<boolean> => {
+  return new Promise(resolve => {
+    const TO_TOPIC = zeroPadValue(watchAddress.toLowerCase(), 32);
+    const filter = {
+      topics: [TRANSFER, null, TO_TOPIC],
+    };
+
+    console.log(
+      `${logPrefix} Watching for ERC-20 transfers to: ${watchAddress} with amount: ${expectedAmount}`,
+    );
+
+    let transferFound = false;
+    let timeoutId: NodeJS.Timeout;
+    let listeners: Array<{ event: any; listener: any }> = [];
+
+    const cleanup = () => {
+      if (timeoutId) clearTimeout(timeoutId);
+      listeners.forEach(({ event, listener }) => {
+        provider.off(event, listener);
+      });
+      listeners = [];
+    };
+
+    const transferListener = (log: any) => {
```

Comment:
I keep reading this as "take the transfer action on a listener" instead of "listen for a transfer".  Please rename to be more clear?